app/services/manifest_builder_v3/canvas_rendering_builder.rb

```diff
@@ -45,7 +45,11 @@ def resource
       # Determines if the resource be downloaded by the user
       # @return [Boolean]
       def downloadable?
-        @record.respond_to?(:resource) || @record.is_a?(FileSet)
+        if @record.respond_to?(:resource) || @record.is_a?(FileSet)
+          @record.resource.decorate.downloadable?
```

Comment:
Can you explain the change in logic here? I think this would be quite expensive, re-fetching the parent for every FileSet.

Reply:
We can't (once we upgrade UV or in the places where we use clover) use the custom UV disableUI service. Most of the download rendering are on the individual canvases so we need to check if we should build them or not. I'll push up a commit that's more efficient.